src/app/screens/swap/useAlexSponsoredTransaction.ts

```diff
@@ -34,36 +27,16 @@ export const useAlexSponsoredTransaction = (userOverrideSponsorValue: boolean) =
   const sponsorTransaction = async (signed: StacksTransaction) =>
     alexSDK.broadcastSponsoredTx(signed.serialize().toString('hex'));
 
-  const getAccountCurrentNonce = async () => {
-    let nonce = 0;
-    const confirmedTransactions: StxTransactionListData = await getConfirmedTransactions({
-      stxAddress,
-      network: selectedNetwork,
-    });
-
-    if (confirmedTransactions && confirmedTransactions.transactionsList.length > 0) {
-      nonce = confirmedTransactions.transactionsList[0].nonce + 1;
-    }
-    const currentPendingNonce =
-      (stxPendingTxData?.pendingTransactions ?? []).reduce(
-        (maxNonce, transaction) => Math.max(maxNonce, transaction?.nonce ?? 0),
-        0,
-      ) + 1;
-
-    return Math.max(currentPendingNonce, nonce);
-  };
-
-  useEffect(() => {
-    (async () => {
-      const currentNonce = await getAccountCurrentNonce();
-      const filteredStxPendingTxData = stxPendingTxData?.pendingTransactions.filter(
-        (transaction) => currentNonce - transaction?.nonce === 1,
-      );
-
-      setHasPendingTransactions(filteredStxPendingTxData?.length > 0);
-    })();
-  }, [stxPendingTxData]);
-
+  const { data: stxPendingTxData } = useStxPendingTxData();
+  const pendingTransactionNonce =
+    stxPendingTxData?.pendingTransactions && stxPendingTxData?.pendingTransactions[0]?.nonce + 1;
```

Comment:
is the order of pendingTransactions here guaranteed? I would be a bit more explicit and take the highest nonce from pending transactions, if that should be the logic.

also the variable name pendingTransactionNonce would be more accurate if you moved the `+ 1` down to L35

Reply:
You’re right about the pending transaction order . I have updated it so the maximum nonce value is catered for when there are multiple pending transactions in different order. 

Although I’m confused as to how adding +1 to line 35 would make a difference? The current nonce would be one value greater than the previous nonce broadcasted. So if you made a transaction with nonce 19 then the user nonce would be 20 since 20 will set as the value for the next transaction. If you calculate the nonce from the confirmed transactions (which we don’t do anymore since we simply get the value from store) but if you have to, you would get the latest confirmed transaction and add one to it. So to get the current pending tx nonce we also do the same that get the pending transaction nonce and add one to it. 
If we don’t then in case of an actual pending transaction, the stxnoxce is 20 and the pending tx nonce will be 19 and since stx nonce>pendingTransactionNonce the pending state would be ignored even though it should not have been

Reply:
thanks for the max nonce fix.

regarding the variable name: it's a case of semantics and readability. not a merge blocker, but to clarify what I meant:

> The current nonce would be one value greater than the previous nonce broadcasted.
actually there is no 'current nonce', just like there is no 'current transaction'. 
there is the previous nonce (previous transactions) and the possible 'next nonce' for future transactions. stxNonce is 'possible next nonce'

> So if you made a transaction with nonce 19 then the user nonce would be 20 since 20 will be set as the value for the next transaction.
the user nonce is not a thing. we calculate 20 but it could be rejected if another transaction gets broadcast with 20 first.

> If you calculate the nonce from the confirmed transactions (which we don’t do anymore since we simply get the value from store) but if you have to, you would get the latest confirmed transaction and add one to it.
yes, this would be determining the possible next nonce

Reply:
totally agree with what you're saying. I have updated the variable name from `pendingTransactionNonce` to `upcomingPendingTransactionNonce` for readability sake